src/oic/oic/provider.py

```diff
@@ -1634,11 +1634,15 @@ def read_registration(self, authn, request, **kwargs):
             return error_response('invalid_request')
         token = authn[len("Bearer "):]
 
-        client_id = self.cdb[token]
+        try:
+            client_id = self.cdb[token]
+        except KeyError:
+            return error_response('invalid_request')
 
         # extra check
         _info = parse_qs(request)
-        assert _info["client_id"][0] == client_id
+        if not _info["client_id"][0] == client_id:
+            return error_response('invalid_request')
```

Comment:
https://github.com/OpenIDC/pyoidc/blob/master/src/oic/extension/provider.py#L466 has similar code (for a more fully featured registration info), and raises Unauthorized() here.

I think Unauthorized is more appropriate for this, as the Bearer token is wrong/non-existing or the client_id is wrong.

Reply:
Changed both of them.